### PR TITLE
Wdp200303 21

### PR DESCRIPTION
**Pull request by AnnaKuszal**
### Stars in ProductBox show by default the average rating: 'stars' from initialState.

*Task: getting opinion by user should remove default rating view:
- Export element 'Stars' to a separate component;
- Added styles - ALL ICONS active on hover, orange color;
- On hover: indicated star and all previous icons become color-filled icon, the rest - empty icon;
- Clicking on any star save results, icon id is saved in component state;
*Saving user rating to app state as 'rating':
- Added Container, defined actions - in result action is achieved: action and changed app state is displayed in Redux devTools;
- in initialState: default rating has been changed for the first product to simulate the feedback for client;

image

### CODE REVIEW

**Comment by Varenthein** [comment]
@jurecki  Review request :)

**Comment by jurecki** [comment]
Generally it's ok, but... in the first product you have user rating, why stars have black background. 
It look like default 'stars"

File: src/components/features/BoxStars/BoxStars.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+              onMouseOver={() => this.onStars(i.id)}
+              icon={
+                (i.active
+                ? i.active === 'active'
+                : i.id <= (this.state.opinion || this.props.rating || stars))
+                  ? faStar
+                  : farStar
+              }
+              className={
+                i.active
```
Comment (jurecki) [inline src/components/features/BoxStars/BoxStars.js]: maybe at the beginning it's worth checking this.props.raiting!==0

**Comment by Varenthein** [comment]
Please fix conflicts

File: src/components/features/BoxStars/BoxStars.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            onClick={e => {
+              e.preventDefault();
+              this.setStars(e, i.id);
+            }}
+          >
+            <FontAwesomeIcon
+              onMouseOver={() => this.onStars(i.id)}
+              icon={
+                (i.active
+                ? i.active === 'active'
```
[the remark this answers is not in the document]
Comment (AnnaKuszal) [reply src/components/features/BoxStars/BoxStars.js]: Yes, for that code with no changes in this.state the comment is absulutely right:)
However, onMouseOver function onStars - line 43 - at first _changes i.active & give them 1 of 2 values ('active' or 'inactive')_ so the condition gets true anyway; And onMouseOut i.active gets 'false' again;

File: src/components/features/BoxStars/BoxStars.js (hunk 1 above)
Comment (AnnaKuszal) [reply src/components/features/BoxStars/BoxStars.js]: It's good to do so... Yet, I haven't insert this for the reason of product content (see initialState) & separating 2 props for 2 different data: 
- stars are set by default as average opinion of all users (value is calculated somehow & based on users cliks);
- rating is supposed to be always 0 by default and only gets changed after click event and reducer action;

**Comment by AnnaKuszal** [comment]
Resolved conflicts.

**Pull request merged**